pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,319 @@
+                    repo = self.get_instance_repo(ns)
+                else:
+                    assert False
+                print('Namespace %s %s' % (ns, repo.len()))
+                qs = [q for q in repo.iter()]
+                for q in qs:
+                    print(q)
+
+        print('NAMESPACES %s' % self._repository.keys())
+        print ('QUALIFIERS')
+        print_repo('qualifier')
+        print('CLASSES')
+        print_repo('class')
+        print('INSTANCES')
+        print_repo('instance')
+
+    def get_class_repo(self, namespace):
+        """
+        Gets the handle for the data CIM class data store for the namespace
+        provided.
+
+          Parameters:
+
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`
+          Returns:
+            returns an instance of InMemoryObjStore which defines the
+            methods exists(), get() create(), etc. for accessing the
+            data in this store.
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+        self.validate_namespace(namespace)
+        return self._repository[namespace]['classes']
+
+    def get_instance_repo(self, namespace):
+        """
+        Gets the handle for the data CIM instance data store for the namespace
+        provided.
+
+          Parameters:
+
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`
+          Returns:
+            returns an instance of InMemoryObjStore which defines the
+            methods exists(), get() create(), etc. for accessing the
+            data in this store.
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+        self.validate_namespace(namespace)
+        return self._repository[namespace]['instances']
+
+    def get_qualifier_repo(self, namespace):
+        """
+        Gets the handle for the data CIM qualifier declaration data store for
+        the namespace provided.
+
+          Parameters:
+
+            namespace (:term:`string`):
+              The name of the CIM namespace in the mock repository. Must not be
+              `None`
+          Returns:
+            returns an instance of InMemoryObjStore which defines the
+            methods exists(), get() create(), etc. for accessing the
+            data in this store.
+        """
+
+        assert namespace
+        namespace = namespace.strip('/')
+        self.validate_namespace(namespace)
+        return self._repository[namespace]['qualifiers']
+
+    def validate_namespace(self, namespace):
+        """
+        If the namespace exists in the store return.
+
+        Raises:
+
+            TODO
+        """
+
+        namespace = namespace.strip('/')
+        try:
+            self._repository[namespace]
+        except KeyError:
+            raise
```

Comment:
In other places in these modules, we have raised KeyError with a short message, instead of re-raising it.